### ISSUE TITLE
Will this repo be maintained anymore?

### ISSUE THREAD

**Issue by MrPowers** [RedisLabs/spark-redis#393]
First off, thank you for building this important library for the Spark Community!

I see that the last commit for this repo was two years ago and there are open issues to add support for [Spark 3.4](https://github.com/RedisLabs/spark-redis/issues/386) and [Spark 3.5](https://github.com/RedisLabs/spark-redis/issues/390).

Is there a project maintainer that would be willing to merge PRs and make releases for new Spark versions?

**Comment by banker** [comment]
@MrPowers Of course! And yes, we'd be happy to look over PRs and release a new version. What's your biggest need right now?

**Comment by MrPowers** [comment]
Thanks @banker!  Would you be willing to push up artifacts that are Spark 3.4 & Spark 3.5 compatible?  Would you be willing to refactor to make spark-redis Spark Connect compatible?

**Comment by mpasa** [comment]
Also, as Spark 4.0 is dropping support for Scala 2.12, it would be great to see it cross-compiled for both 2.12 and 2.13. This is currently blocking our version upgrade and we'd need to drop the library if not possible.

**Comment by newfront** [comment]
I added a branch for 4.0 (https://github.com/RedisLabs/spark-redis/pull/394). @banker there is one open question regarding expected behavior for the `RedisNode.idx` index. With the modifications to Jedis they deprecated `conn.clusterSlots` in favor of `conn.clusterShards` which provides different behavior. All cluster slots per Node are returned - so this might change the way things work. If you have a chance to look things over I'd be more than happy to change the logic.

**Comment by ialarmedalien** [comment]
@banker @newfront what is the status of the migration to spark 4.0? Is this project still active? There doesn't seem to have been any change of consequence since late 2024.

**Comment by newfront** [comment]
@ialarmedalien I left (https://github.com/RedisLabs/spark-redis/pull/394) as a partial WIP. I needed to understand what is expected in cluster mode but couldn't understand what "worked" before since the underlying Jedis implementation modifies the resp bodies to converge two calls in one now.

The unit tests all passed when I opened up the PR last year, with official Spark 4 now out the door there is probably a little more cleanup work to get things inline.

**Comment by llphxd** [comment]
Hi — this repo has been quiet for a long time, and the original connector was built mainly around RDD / older Spark APIs, so it often breaks or feels awkward on Spark 3.x + Spark SQL.
On top of that, its Redis data type coverage is incomplete and inconsistent: some types are only partially supported, schema mapping is not always clear, and batch read/write behavior can differ between types. If you need predictable, type-aware access across common Redis structures, that gap shows up quickly.
If your use case is **batch read/write with Spark SQL or DataFrame only** (not RDD / Streaming), you might want to look at a community project I maintain:
**https://github.com/llphxd/spark-redis-connector**
It is a DataSource V2 connector focused on `USING redis` / `.format("redis")`, with batch support for **string, hash, list, set, and sorted set**, each exposed as a clear table layout rather than legacy RDD wrappers.
